packages/core/src/application_init.ts

```diff
@@ -27,6 +27,80 @@ import {noop} from './util/noop';
  *
  * @see `ApplicationInitStatus`
  *
+ * @usageNotes
+ *
+ * ### Usage examples
+ *
+ * The following module provides a function returning a promise to the `APP_INITIALIZER` token.
+ *
+ * ```
+ *  function initializeApp(): Promise<any> {
+ *    return new Promise((resolve, reject) => {
+ *      // Do some asynchronous stuff
+ *      resolve();
+ *    });
+ *  }
+ *
+ *  @NgModule({
+ *   imports: [BrowserModule],
+ *   declarations: [AppComponent, HelloComponent],
```

Comment:
I believe we can drop `HelloComponent` here (and in other example below)?
```suggestion
 *   declarations: [AppComponent],
```